src/mpl/src/gpu/mpl_gpu_cuda.c

```diff
@@ -0,0 +1,75 @@
+/*
+ *  Copyright (C) by Argonne National Laboratory.
+ *      See COPYRIGHT in top-level directory.
+ */
+
+#include "mpl.h"
+
+MPL_SUPPRESS_OSX_HAS_NO_SYMBOLS_WARNING;
+
+#ifdef MPL_HAVE_CUDA
+
+#define CUDA_ERR_CHECK(ret) if (unlikely((ret) != cudaSuccess)) goto fn_fail
+
+int MPL_gpu_query_pointer_type(const void *ptr, MPL_pointer_type_t * attr)
+{
+    cudaError_t ret;
+    struct cudaPointerAttributes ptr_attr = { 0 };
+    ret = cudaPointerGetAttributes(&ptr_attr, ptr);
+    CUDA_ERR_CHECK(ret);
+    switch (ptr_attr.type) {
+        case cudaMemoryTypeUnregistered:
+        case cudaMemoryTypeHost:
+            *attr = MPL_GPU_POINTER_HOST;
+            break;
+        case cudaMemoryTypeDevice:
+            *attr = MPL_GPU_POINTER_DEV;
+            break;
+        case cudaMemoryTypeManaged:
+            *attr = MPL_GPU_POINTER_MANAGED;
+            break;
+    }
```

Comment:
This check is only valid for CUDA version >= 11.  For older versions, you'll need to check for the error case and map that to `cudaMemoryTypeUnregistered`.

Reply:
MIssed this one the last push, fixing it now.